packages/runtime/container-runtime/src/test/gc/gc.perf.spec.ts

```diff
@@ -0,0 +1,227 @@
+	IGarbageCollector,
+	IGarbageCollectorCreateParams,
+	stableGCVersion,
+} from "../../gc/index.js";
+import { ContainerRuntimeGCMessage } from "../../messageTypes.js";
+import { pkgVersion } from "../../packageVersion.js";
+import { createTestConfigProvider } from "./gcUnitTestHelpers.js";
+
+type GcWithPrivates = IGarbageCollector & {
+	baseSnapshotDataP: Promise<IGarbageCollectionSnapshotData | undefined>;
+	initializeOrUpdateGCState: () => Promise<void>;
+};
+
+function createGarbageCollector(params: {
+	deleteSweepReadyNodes: (sweepReadyRoutes: string[]) => string[];
+	getGCData: () => Promise<IGarbageCollectionData>;
+	logger: MockLogger;
+	createParams?: Partial<IGarbageCollectorCreateParams>;
+	gcBlobsMap?: Map<string, any>;
+	gcMetadata?: IGCMetadata;
+	closeFn?: (error?: ICriticalContainerError) => void;
+	isSummarizerClient?: boolean;
+}): GcWithPrivates {
+	const {
+		createParams = {},
+		gcBlobsMap = new Map(),
+		gcMetadata = {},
+		closeFn = () => {},
+		isSummarizerClient = true,
+		getGCData,
+		deleteSweepReadyNodes,
+		logger,
+	} = params;
+	const testPkgPath = ["testPkg"];
+
+	const getNodeType = (nodePath: string) => {
+		if (nodePath.split("/").length !== 2) {
+			return GCNodeType.Other;
+		}
+		return GCNodeType.DataStore;
+	};
+
+	// The runtime to be passed to the garbage collector.
+	const gcRuntime: IGarbageCollectionRuntime = {
+		updateStateBeforeGC: async () => {},
+		getGCData,
+		updateUsedRoutes: (usedRoutes: string[]) => {
+			return { totalNodeCount: 0, unusedNodeCount: 0 };
+		},
+		deleteSweepReadyNodes,
+		updateTombstonedRoutes: (tombstoneRoutes: string[]) => {},
+		getNodeType,
+		getCurrentReferenceTimestampMs: () => Date.now(),
+		closeFn,
+	};
+
+	let metadata = createParams.metadata;
+	const existing = createParams.baseSnapshot !== undefined;
+	// For existing, add container runtime metadata which is required for GC to be enabled.
+	if (existing) {
+		metadata = {
+			...metadata,
+			...gcMetadata,
+			gcFeature: gcMetadata.gcFeature ?? stableGCVersion,
+			summaryFormatVersion: 1,
+			message: undefined,
+		};
+	}
+
+	return GarbageCollector.create({
+		...createParams,
+		runtime: gcRuntime,
+		gcOptions: createParams.gcOptions ?? {},
+		baseSnapshot: createParams.baseSnapshot,
+		baseLogger: createChildLogger({ logger }),
+		existing,
+		metadata,
+		createContainerMetadata: {
+			createContainerRuntimeVersion: pkgVersion,
+			createContainerTimestamp: Date.now(),
+		},
+		isSummarizerClient,
+		readAndParseBlob: async <T>(id: string) => gcBlobsMap.get(id) as T,
+		getNodePackagePath: async (nodeId: string) => testPkgPath,
+		getLastSummaryTimestampMs: () => Date.now(),
+		submitMessage: (message: ContainerRuntimeGCMessage) => {},
+		sessionExpiryTimerStarted: createParams.sessionExpiryTimerStarted,
+	}) as GcWithPrivates;
+}
+
+describe("GC benchmark tests", () => {
+	/**
+	 * Called when sweep runs. It deleted the nodes from gcData.
+	 */
+	function deleteSweepReadyNodes(
```

Comment:
Why does this need to be implemented?

Reply:
Good point. I removed a bunch of things that were unncessary (copy pasted from another test)